src/deepEqual.js

```diff
@@ -21,6 +21,37 @@
  */
 function deepEqual(a, b) {
   // write code here
+  let typeOfA = typeof a;
+  let typeOfB = typeof b;
+
+  if (typeOfA === typeOfB && typeOfA !== null && typeOfB !== null) {
+    if (typeOfA === 'object') {
+      if (!isEqualObjectsByKeys(a, b)) {
+        return false;
+      }
+      if (!isEqualObjectsByKeys(b, a)) {
+        return false;
+      }
```

Comment:
figure out how to fix isEqualObjectsByKeys so that you don't have to call it twice.

Reply:
Done)